readthedocs/rtd_tests/utils.py

```diff
@@ -92,6 +79,36 @@ def make_test_git():
     return directory
 
 
+@restoring_chdir
+def add_submodule_without_cloning(directory, submodule, url):
```

Comment:
Could you add the parameters in the docstring also? Like,

```
:param directory: path to the git directory
:type directory: str

...

:returns:
:rtype:
```

Also, maybe is good to add `git` in the function's name. Like, `add_git_submodule_without_cloning_it` or `add_git_submodule_to_index_directly` or ...